### DISCUSSION TITLE
Graviton2 supports the Large System Extensions

### DISCUSSION THREAD

**Discussion by EricDunaway** [cargo-lambda/cargo-lambda#610] [Q&A] [answered]
I saw that Gravitron 2 processors support LSE and some documentation recommended `RUSTFLAGS` include `-Ctarget-feature=+lse`. Skimming the code for the project I found the [section where the target CPU is set](https://github.com/cargo-lambda/cargo-lambda/blob/7807a975528668260b132b1c33dbbd91a5a62861/crates/cargo-lambda-build/src/lib.rs#L175-L176) will pass through any RUSTFLAGS set but does not currently set the lse feature. 

If it is out of scope for this project to include the lse feature, is there a recommended way to set RUSTFLAGS for architecture-specific builds other than just setting them before calling cargo?

Has anyone already benchmarked the effect of LSE on tokio or other crates? Maybe core counts in lambdas don't get high enough for it to make an impact.

Links:
  - https://github.com/aws/aws-graviton-getting-started/blob/main/rust.md
  - https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-to-arm-based-aws-graviton2-processors/
  - https://learn.arm.com/learning-paths/servers-and-cloud-computing/lse/intro/
    - > One of the common performance issues when migrating to Neoverse is running software that does not utilize LSE. Software built with load exclusives and store exclusives usually runs slower on Neoverse instances.

**Comment by calavera** [comment, marked as answer]
I didn't know that flag existed, this just why it's not set. You can experiment with it by running those same commands that the graviton-getting-started indicates:

```
export RUSTFLAGS="-Ctarget-feature=+lse"
cargo lambda build --release
```

If it's useful, feel free to open a PR to add the flag in the same place that we have the `target-cpu` flag, the logic should be pretty similar.
